wdl2cwl/tests/oop_cwl_files/bcftools_stats.cwl

```diff
@@ -0,0 +1,170 @@
+  - id: regions
+    type:
+      - string
+      - 'null'
+  - id: regionsFile
+    type:
+      - File
+      - 'null'
+  - id: samplesFile
+    type:
+      - File
+      - 'null'
+  - id: targets
+    type:
+      - string
+      - 'null'
+  - id: targetsFile
+    type:
+      - File
+      - 'null'
+  - id: userTsTv
+    type:
+      - string
+      - 'null'
+  - id: threads
+    default: 0
+    type: int
+  - id: memory
+    default: 256M
+    type: string
+  - id: dockerImage
+    default: quay.io/biocontainers/bcftools:1.10.2--h4f4756c_2
+    type: string
+outputs:
+  - id: stats
+    type: File
+    outputBinding:
+        glob: "$(inputs.outputPath === null ? (inputs.inputVcf.basename + '.stats')\
+            \ : inputs.outputPath)"
+requirements:
+  - class: DockerRequirement
+    dockerPull: quay.io/biocontainers/bcftools:1.10.2--h4f4756c_2
+  - class: InitialWorkDirRequirement
+    listing:
+      - entryname: example.sh
+        entry: |4
+
+            set -e
+            mkdir -p \$(dirname  $(inputs.outputPath === null ? inputs.inputVcf.basename + '.stats' : inputs.outputPath))
+            bcftools stats \
+            $(inputs.afBins === null ? "" : "--af-bins " inputs.afBins) \
+            $(inputs.afTag === null ? "" : "--af-tag " inputs.afTag) \
+            $(inputs.firstAlleleOnly ? "--1st-allele-only" : "") \
+            $(inputs.collapse === null ? "" : "--collapse " inputs.collapse) \
+            $(inputs.depth === null ? "" : "--depth " inputs.depth) \
+            $(inputs.exclude === null ? "" : "--exclude " inputs.exclude) \
+            $(inputs.exons === null ? "" : "--exons " inputs.exons) \
+            $(inputs.applyFilters === null ? "" : "--apply-filters " inputs.applyFilters) \
+            $(inputs.fastaRef === null ? "" : "--fasta-ref " inputs.fastaRef) \
+            $(inputs.include === null ? "" : "--include " inputs.include) \
+            $(inputs.splitByID ? "--split-by-ID" : "") \
+            $(inputs.regions === null ? "" : "--regions " inputs.regions) \
+            $(inputs.regionsFile === null ? "" : "--regions-file " inputs.regionsFile) \
+            $(inputs.samples.map(function(el) {return el.path}).join(",")) \
+            $(inputs.samplesFile === null ? "" : "--samples-file " inputs.samplesFile) \
+            $(inputs.targets === null ? "" : "--targets " inputs.targets) \
+            $(inputs.targetsFile === null ? "" : "--targets-file " inputs.targetsFile) \
+            $(inputs.userTsTv === null ? "" : "--user-tstv " inputs.userTsTv) \
+            --threads $(inputs.threads) \
+            $(inputs.verbose ? "--verbose" : "") \
+            $(inputs.inputVcf)$(inputs.compareVcf) > $(inputs.outputPath === null ? inputs.inputVcf.basename + '.stats' : inputs.outputPath)
+  - class: InlineJavascriptRequirement
+  - class: NetworkAccess
+    networkAccess: true
+  - class: ResourceRequirement
+    coresMin: $(inputs.threads + 1)
+  - class: ResourceRequirement
+    ramMin: |-
```

Comment:
There should be a single `ResourceRequirement`, with both `coresMin` and `ramMin`

Reply:
This isn't consistent with other cwl files. How do I achieve this?

Reply:
Each `- class: ResourceRequirement` is an YAML node/object. `requirements` is a collection/list (see how each of its children in this YAML start with a dash `-`?).

So I think here we have two entries in this list that define `- class: ResourceRequirement`, which in CWL would mean you are defining that twice. Not sure if it works, probably depends on implementation?

Anyway, there's an example here in the user guide with both `coreMin` and `ramMin` in the same `ResourceRequirement` object: https://www.commonwl.org/user_guide/20-software-requirements/index.html

If you'd like to see how the YAML document is parsed, @Th3nn3ss , you can try this. Go to this URL <https://codebeautify.org/yaml-parser-online>. Then paste this simplified version of this YAML.

```yaml
class: CommandLineTool
# simplify
requirements:
  - class: ResourceRequirement
    coresMin: $(inputs.threads + 1)
  - class: ResourceRequirement
    ramMin: |-
        ${
        var unit = inputs.memory.match(/[a-zA-Z]+/g).join("");
        }
```

It should show the two `ResourceRequirement`'s under `requirements (we want one). You can play with that online editor and try other things to see what the YAML output will be.

![image](https://user-images.githubusercontent.com/304786/147294529-8726ae06-257b-4e74-b133-1e80c0017f30.png)